rest_client/client.py

```diff
@@ -0,0 +1,96 @@
+import uuid
+from typing import Literal
+from urllib.parse import urljoin
+
+import curlify
+import structlog
+from requests import session
+from requests.exceptions import JSONDecodeError
+
+from rest_client.configuration import Configuration
+
+HttpMethod = Literal["GET", "POST", "PUT", "DELETE"]
+
+
+class RestClient:
+
+    def __init__(self, configuration: Configuration):
+        self.host = configuration.host
+        self.headers = configuration.headers
+        self.disable_log = configuration.disable_log
+        self.session = session()
+        self.log = structlog.getLogger(__name__).bind(service='api')
+
+    def get(self, path: str, **kwargs):
+        """
+        GET request
+        :param path: endpoint
+        :param kwargs: остальные параметры
+        :return: response
+        """
+        return self._send_request(method="GET", path=path, **kwargs)
+
+    def post(self, path: str, **kwargs):
+        """
+          POST request
+          :param path: endpoint
+          :param kwargs: остальные параметры
+          :return: response
+          """
+        return self._send_request(method="POST", path=path, **kwargs)
+
+    def put(self, path: str, **kwargs):
+        """
+          PUT request
+          :param path: endpoint
+          :param kwargs: остальные параметры
+          :return: response
+          """
+        return self._send_request(method="PUT", path=path, **kwargs)
+
+    def delete(self, path: str, **kwargs):
+        """
+          DELETE request
+          :param path: endpoint
+          :param kwargs: остальные параметры
+          :return: response
+          """
+        return self._send_request(method="DELETE", path=path, **kwargs)
+
+    def _send_request(self, method: HttpMethod, path: str, **kwargs):
```

Comment:
Аннотации типов хорошая практика, чтобы совсем уж было красиво, можно все тогда докрутить
```suggestion
    def _send_request(self, method: HttpMethod, path: str, **kwargs: Any) -> Response:
```